app/src/main/java/com/tananaev/logcat/LineAdapter.java

```diff
@@ -30,7 +43,11 @@
     private List<Line> linesAll = new LinkedList<>();
     private List<Line> linesFiltered = new LinkedList<>();
 
-    private String keyword;
+    private String tag = "";
```

Comment:
I think default should be `null`, not empty string. Applies to all variables here.